libpod/shimv2.go

```diff
@@ -0,0 +1,776 @@
+		return err
+	}
+
+	return nil
+}
+
+func (r *shimv2) PauseContainer(ctr *Container) error {
+	return fmt.Errorf("not supported")
+}
+
+func (r *shimv2) UnpauseContainer(ctr *Container) error {
+	return fmt.Errorf("not supported")
+}
+
+func (r *shimv2) HTTPAttach(ctr *Container, req *http.Request, w http.ResponseWriter, streams *HTTPAttachStreams, detachKeys *string, cancel <-chan bool, hijackDone chan<- bool, streamAttach, streamLogs bool) (deferredErr error) {
+	return fmt.Errorf("not supported")
+}
+
+func (r *shimv2) AttachResize(ctr *Container, newSize define.TerminalSize) error {
+	return fmt.Errorf("not supported")
+}
+
+func (r *shimv2) CheckpointContainer(ctr *Container, options ContainerCheckpointOptions) error {
+	return fmt.Errorf("not supported")
+}
+
+// since conmon is not needed here, let's mock the response as true.
+// TODO: the shimv2 PID could be checked to get its state
+func (r *shimv2) CheckConmonRunning(ctr *Container) (bool, error) {
+	return true, nil
+}
+
+func (r *shimv2) SupportsCheckpoint() bool {
+	return false
+}
+
+func (r *shimv2) SupportsJSONErrors() bool {
+	return false
+}
+
+func (r *shimv2) SupportsNoCgroups() bool {
+	return false
+}
+
+func (r *shimv2) SupportsKVM() bool {
+	return r.supportsKVM
+}
+
+func (r *shimv2) ExitFilePath(ctr *Container) (string, error) {
+	if ctr == nil {
+		return "", errors.Wrapf(define.ErrInvalidArg, "must provide a valid container to get exit file path")
+	}
+	return filepath.Join(r.exitsDir, ctr.ID()), nil
+}
+
+func (r *shimv2) RuntimeInfo() (*define.ConmonInfo, *define.OCIRuntimeInfo, error) {
+	return nil, nil, fmt.Errorf("not supported")
+}
+
+func (r *shimv2) ExecContainerHTTP(ctr *Container, sessionID string, options *ExecOptions, req *http.Request, w http.ResponseWriter, streams *HTTPAttachStreams, cancel <-chan bool, hijackDone chan<- bool, holdConnOpen <-chan bool, newSize *define.TerminalSize) (int, chan error, error) {
+	return 0, nil, fmt.Errorf("not supported")
+}
+
+func (r *shimv2) ExecContainer(c *Container, sessionID string, options *ExecOptions, streams *define.AttachStreams, newSize *define.TerminalSize) (int, chan error, error) {
+	return 0, nil, fmt.Errorf("not supported")
+}
+
+func (r *shimv2) ExecContainerDetached(ctr *Container, sessionID string, options *ExecOptions, stdin bool) (int, error) {
+	return 0, fmt.Errorf("not supported")
+}
+
+func (r *shimv2) ExecAttachResize(ctr *Container, sessionID string, newSize define.TerminalSize) error {
+	return fmt.Errorf("not supported")
+}
+
+func (r *shimv2) ExecStopContainer(ctr *Container, sessionID string, timeout uint) error {
+	return fmt.Errorf("not supported")
+}
+
+func (r *shimv2) ExecUpdateStatus(ctr *Container, sessionID string) (bool, error) {
+	return false, fmt.Errorf("not supported")
+}
+
+func (r *shimv2) ExecAttachSocketPath(ctr *Container, sessionID string) (string, error) {
+	return "", fmt.Errorf("not supported")
+}
+
+func (r *shimv2) UpdateContainerStatus(ctr *Container) error {
```

Comment:
From my understanding, when the shimv2 daemon is started there is one argument that can be set, this is a ttrpc server exposed by containerd and it is used by the shimv2 to publish events back.

I tried to mimic what cri-o was doing since both rely on conmon, and this is not being set when kata v2 is used. I will try to understand more about how cri-o receives kata container events